frontend/src/components/MapTooltip/MapTooltip.tsx

```diff
@@ -10,43 +14,90 @@ type MapTooltipProps = {
   coordinates: LngLat
 }
 
-let MapTooltip: React.StatelessComponent<MapTooltipProps> = ({ children, coordinates }) =>
-  <Popup
-    anchor='bottom'
-    className='MapTooltip'
-    coordinates={coordinates.toArray()}
-    offset={[0, -20]}
-  >
-    <table>
-      <tbody>
-        {children}
-      </tbody>
-    </table>
-  </Popup>
-
-type Props = {
+let MapTooltip: React.StatelessComponent<MapTooltipProps> = ({ children, coordinates }) => {
+  return (
+    <Popup
+      anchor='bottom'
+      className='MapTooltip'
+      coordinates={coordinates.toArray()}
+      offset={[0, -20]}
+    >
+      {children}
+    </Popup>
+  )
+}
+
+type ProviderProps = {
+  store: Store
   point: GeoJSONEventData
 }
 
+type RepresentativePointProps = {
+  point: GeoJSONEventData
+}
+
+function _ToggleNextProvider(store: Store, maxIndex: number) {
+  console.log('toggle next')
+  let index = store.get('providerIndex')
+  if (store.get('providerIndex') === maxIndex)
+    return
+  store.set('providerIndex')(index + 1)
+}
+
+function _TogglePrevProvider(store: Store) {
+  console.log('toggle prev')
+  let index = store.get('providerIndex')
+  if (index === 0)
+    return
+  store.set('providerIndex')(index - 1)
+}
+
+
 /**
  * TODO: expose info for all providers with the same address,
  * not only the first one.
  */
-export let ProviderPopup: React.StatelessComponent<Props> =
-  ({ point: { features, lngLat } }) =>
-    <MapTooltip coordinates={lngLat}>
-      <TableRow name='Address' value={features[0].properties.address} />
-      <TableRow name='Specialty' value={features[0].properties.specialty} />
-      <TableRow name='Lat' value={formatCoordinate(lngLat.lat)} />
-      <TableRow name='Lng' value={formatCoordinate(lngLat.lng)} />
-    </MapTooltip>
+export let ProviderPopup: React.StatelessComponent<ProviderProps> =
+  ({ store, point: { features, lngLat } }) => {
+    console.log('features')
+    let index = store.get('providerIndex')
+    let maxIndex = features.length - 1
+    let indexMessage = (index + 1) + ' of ' + (maxIndex + 1)
+    console.log(maxIndex)
+    return (
+      <MapTooltip coordinates={lngLat}>
+        <table>
+          <tbody>
+            <TableRow name='NPI' value={features[index].properties.npi} />
+            <TableRow name='Address' value={features[index].properties.address} />
+            <TableRow name='Specialty' value={features[index].properties.specialty} />
+            <TableRow name='Lat' value={formatCoordinate(lngLat.lat)} />
+            <TableRow name='Lng' value={formatCoordinate(lngLat.lng)} />
+          </tbody>
+        </table>
+        <div className='controls Flex -Center'>
+          <IconButton touchRippleColor='white' onClick={() => _TogglePrevProvider(store)}>
```

Comment:
@bcherny is there a simple way to abstract the button maybe?

Reply:
This looks pretty good to me. TBH we're fighting MaterialUI with all of its inline styles until they release the new version, so this is more or less as good as it'll get.